cloud/openstack/os_keystone_endpoint.py

```diff
@@ -0,0 +1,206 @@
+     description:
+        - Admin endpoint URL
+     required: false
+     default: None
+   state:
+     description:
+        - Indicate desired state of the resource
+     choices: ['present', 'absent']
+     default: present
+requirements: ["shade"]
+'''
+
+EXAMPLES = '''
+# Create a Glance endpoint in region aw1
+- os_keystone_endpoint:
+    cloud: mycloud
+    state: present
+    service: image
+    region: aw1
+    public_url: https://glance.aw1.bigcompany.com/
+    internal_url: http://glance-aw1.internal:9292/
+    admin_url: https://glance.aw1.bigcompany.com/
+
+# Delete a Glance endpoint in region aw2
+- os_keystone_endpoint:
+    cloud: mycloud
+    state: absent
+    service: image
+    region: aw2
+    public_url: https://glance.aw1.bigcompany.com/
+    internal_url: http://glance-aw1.internal:9292/
+    admin_url: https://glance.aw1.bigcompany.com/
+'''
+
+RETURN = '''
+endpoint:
+    description: Dictionary describing the endpoint.
+    returned: On success when I(state) is 'present'
+    type: dictionary
+    contains:
+        id:
+            description: Endpoint ID.
+            type: string
+            sample: "5eed2b7a088b4ac3b7272a4879317fd4"
+        region:
+            description: Endpoint region.
+            type: string
+            sample: "RegionOne"
+        service_id:
+            description: ID of the service for the endpoint.
+            type: string
+            sample: "33821361fe914df9ab58cbf79c6b5388"
+        adminurl:
+            description: Endpoint admin URL.
+            type: string
+            sample: "http://http://192.168.122.31:35357/v2.0"
+        internalurl:
+            description: Endpoint internal URL.
+            type: string
+            sample: "http://http://192.168.122.31:5000/v2.0"
+        publicurl:
+            description: Endpoint public URL.
+            type: string
+            sample: "http://192.168.122.31:5000/v2.0"
+'''
+
+
+def _system_state_change(state, endpoint):
+    if state == 'present' and not endpoint:
+        return True
+    if state == 'absent' and endpoint:
+        return True
+    return False
+
+
+def main():
+    argument_spec = openstack_full_argument_spec(
+        service=dict(required=True),
+        region=dict(required=False, default=None),
+        public_url=dict(required=True),
+        internal_url=dict(required=False, default=None),
+        admin_url=dict(required=False, default=None),
+        state=dict(default='present', choices=['present', 'absent']),
+    )
+
+    module_kwargs = openstack_module_kwargs()
+    module = AnsibleModule(argument_spec, supports_check_mode=True, **module_kwargs)
+
+    if not HAS_SHADE:
+        module.fail_json(msg='shade is required for this module')
+
+    service = module.params.pop('service')
+    region = module.params.pop('region')
+    public_url = module.params.pop('public_url')
+    internal_url = module.params.pop('internal_url')
+    admin_url = module.params.pop('admin_url')
```

Comment:
In keystone v3 this would be better represented as an individual url with an interface, so you say: 

service: identity 
interface: public 
url: http://the-url/
region: RegionOne